src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -214,12 +216,12 @@ private static AnalysisResults analyzeImage(ByteString imageBytes)
   }
 
   /**
-   * Checks the provided tokens for a date that can be parsed. If one is found, it is added to the
-   * builder as a timestamp.
+   * Checks the raw text in the builder for a date that can be parsed. If one is found, it is added
+   * to the builder as a timestamp.
    */
-  private static void checkForParsableDate(
-      AnalysisResults.Builder analysisBuilder, Stream<String> tokens) {
-    Stream<String> dates = tokens.filter(ReceiptAnalysis::isDate);
+  private static void checkForParsableDate(AnalysisResults.Builder analysisBuilder) {
+    Stream<String> dates =
+        getTokensFromRawText(analysisBuilder.getRawText()).filter(ReceiptAnalysis::isDate);
```

Comment:
Is there a benefit in filtering for `hasDigit` since the `isDate` and `isPrice` filters are applied anyways?